cmd/config-sync/config_sync.go

```diff
@@ -171,3 +202,257 @@ func (c *ConfigSync) syncConfig(desired *qdr.BridgeConfig) error {
 	}
 	return nil
 }
+
+func (c *ConfigSync) syncRouterConfig(desired *qdr.RouterConfig) error {
+	agent, err := c.agentPool.Get()
+	if err != nil {
+		return fmt.Errorf("Could not get management agent : %s", err)
+	}
+	var synced bool
+	for i := 0; i < 3 && err == nil && !synced; i++ {
+		synced, err = syncRouterConfig(agent, desired, c)
+	}
+	c.agentPool.Put(agent)
+	if err != nil {
+		return fmt.Errorf("Error while syncing router config : %s", err)
+	}
+	if !synced {
+		return fmt.Errorf("Failed to sync router config")
+	}
+	return nil
+}
+
+func syncRouterConfig(agent *qdr.Agent, desired *qdr.RouterConfig, c *ConfigSync) (bool, error) {
+	actual, err := agent.GetLocalConnectorStatus()
+	if err != nil {
+		return false, fmt.Errorf("Error retrieving local connector status: %s", err)
+	}
+
+	differences := qdr.ConnectorsDifference(actual, desired)
+	if differences.Empty() {
+		err = c.checkConnectorSecrets(actual, SHARED_TLS_DIRECTORY)
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+	} else {
+
+		err := c.syncConnectorSecrets(differences, SHARED_TLS_DIRECTORY)
+		if err != nil {
+			return false, fmt.Errorf("error syncing secrets: %s", err)
+		}
+
+		if err = agent.UpdateConnectorConfig(differences); err != nil {
+			return false, fmt.Errorf("Error syncing connectors: %s", err)
+		}
+		return false, nil
+	}
+}
+
+func syncSecrets(configSync *ConfigSync, changes *qdr.BridgeConfigDifference, sharedTlsFilesDir string) error {
+	for _, added := range changes.HttpListeners.Added {
+		if len(added.SslProfile) > 0 {
+			log.Printf("Copying cert files related to HTTP Connector sslProfile %s", added.SslProfile)
+			return configSync.ensureSslProfile(added.SslProfile, added.SslProfile, sharedTlsFilesDir)
```

Comment:
If something gets added here, the remaining changes are being ignored.
Maybe should only return at this point in case of an error.